pkgs/by-name/oc/ocrodjvu/package.nix

```
@@ -0,0 +1,77 @@
+{
+  lib,
+  python3Packages,
+  fetchFromGitHub,
+  djvulibre,
+  docbook-xsl-ns,
+  glibcLocales,
+  libxml2,
+  libxml2Python,
+  libxslt,
+  pkg-config,
+  tesseract5,
+  withCuneiform ? false,
+  cuneiform,
+  withGocr ? false,
+  gocr,
+  withOcrad ? false,
+  ocrad,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ocrodjvu";
+  version = "0.13.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "FriedrichFroebel";
+    repo = "ocrodjvu";
+    rev = version;
+    hash = "sha256-EiMCrRFUAJbu9QLgKpFIKqigCZ77lpTDD6AvZuMbyhA=";
+  };
+
+  build-system =
+    with python3Packages;
+    [
+      cython
+      djvulibre
+      docbook-xsl-ns
+      glibcLocales
+      html5lib
+      libxml2
+      libxml2Python
+      libxslt
+      packaging
+      pillow
+      pkg-config
+      pyicu
+      setuptools
+      tesseract5
+      wheel
+    ]
+    ++ lib.optional withCuneiform cuneiform
+    ++ lib.optional withGocr gocr
+    ++ lib.optional withOcrad ocrad;
+
+  dependencies = with python3Packages; [
+    lxml
+    python-djvulibre
+  ];
+
+  nativeCheckInputs = [ python3Packages.unittestCheckHook ];
```

```suggestion
  build-system =
    with python3Packages;
    [
      setuptools
    ]
    ++ lib.optional withCuneiform cuneiform
    ++ lib.optional withGocr gocr
    ++ lib.optional withOcrad ocrad;

  dependencies = with python3Packages; [
    lxml
    python-djvulibre
    pyicu
    html5lib
  ];

  nativeCheckInputs = [
    python3Packages.unittestCheckHook
    python3Packages.pillow
    djvulibre
    glibcLocales
    libxml2
    libxml2Python
    tesseract5
  ];
```
Checked again which packages are required to build / test / run.

---

Oh and you probably want cuneiform, gocr and ocrad in propagatedBuildInputs instead.